content/300-guides/400-migrate-to-prisma/03-migrate-from-mongoose.mdx

```diff
@@ -0,0 +1,1066 @@
+        profile: {
+          set: { bio },
+        },
+      },
+    })
+
+    return res.status(200).json(user)
+  } catch (error) {
+    console.log(error)
+    return res.status(500).json(error)
+  }
+}
+```
+
+### 4.3. Replacing queries in `PUT` requests
+
+The REST API has two routes that accept a `PUT` request:
+
+- `/post/:id/:categoryId`: Adds the post with `:id` to the category with `:categoryId`
+- `/post/:id`: Updates the `published` status of a post to true.
+
+Let's dive into the route handlers that implement these requests.
+
+#### `/post/:id/:categoryId`
+
+The `/post/:id/:categoryId` handler is currently implemented as follows:
+
+```js file=src/controllers/post.js
+const addPostToCategory = async (req, res) => {
+  const { id, categoryId } = req.params
+
+  try {
+    const category = await Category.findById(categoryId)
+
+    if (!category)
+      return res.status(404).json({ message: 'Category not found' })
+
+    const post = await Post.findByIdAndUpdate(
+      { _id: id },
+      {
+        categories: [{ _id: categoryId }],
+      },
+      { new: true }
+    )
+
+    if (!post) return res.status(404).json({ message: 'Post not found' })
+    return res.status(200).json(post)
+  } catch (error) {
+    return res.status(500).json(error)
+  }
+}
+```
+
+With Prisma, the handler is implemented as follows:
+
+```js file=src/controllers/post.js
+const addPostToCategory = async (req, res) => {
+  const { id, categoryId } = req.query
+
+  try {
+    const post = await prisma.post.update({
+      where: {
+        id,
+      },
+      data: {
+        categories: {
+          connect: {
+            id: categoryId,
+          },
+        },
+      },
+    })
+
+    if (!post) return res.status(404).json({ message: 'Post not found' })
+
+    return res.status(200).json(post)
+  } catch (error) {
+    console.log({ error })
+    return res.status(500).json(error)
+  }
+}
+```
+
+#### `/post/:id`
+
+The `/post/:id` handler is currently implemented as follows:
+
+```js file=src/controllers/post.js
+const publishDraft = async (req, res) => {
+  const { id } = req.params
+
+  try {
+    const post = await Post.findByIdAndUpdate(
+      { id },
+      { published: true },
+      { new: true }
+    )
+    return res.status(200).json(post)
+  } catch (error) {
+    return res.status(500).json(error)
+  }
+}
+```
+
+With Prisma, the handler is implemented as follows:
+
+```js file=src/controllers/post.js
+const publishDraft = async (req, res) => {
+  const { id } = req.params
+
+  try {
+    const post = await prisma.post.update({
+      where: { id },
+      data: { published: true },
+    })
+    return res.status(200).json(post)
+  } catch (error) {
+    return res.status(500).json(error)
+  }
+}
+```
+
+## More
+
+### Embedded documents `_id` field
+
+By default, Mongoose assigns each document and embedded document an `_id` field. If you wish to disable this option for embedded documents, you can set the `_id` option to false.
```

Comment:
In a migration scenario, why and when wold they want to disable this option? Is it something they might want to do before migrating, if they don't want to migrate those IDs fro the embedded documents? The way it reads now, it sounds like an option for people that are staying with Mongoose, not migrating.

Reply:
This is entirely optional and its existence/ absence doesn't affect the migration process. Prisma will include it for you in your application. I think the "More" section was intended to document what the library the user is migrating from.

The other documents include an `_id`.